c10/util/constexpr_string_functions.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <cstdlib>
+
+namespace c10 {
+namespace util {
+
+// In general, constexpr algorithms in C++ must be written recursively,
+// as looping constructs (like for) are not constexpr.  This code is
+// written using ternaries as a holdover from C++11; but as of C++14
+// constexpr functions are allowed to have multiple return statements
+// (writing them with ternaries is more portable though!)
```

Comment:
Quick thought, is there an existing API for this kind of stuff we could emulate, like in Boost or something? (maybe you are already?)